### PR TITLE
Removing vulnerabilities, code smells in test-scenario module

### PR DESCRIPTION
**Pull request by yesamer**
No changes on functionalities, I just resolved all reported vulnerabilities (17) by sonar for `drools-scenario-simulation` plus some code smells.

**referenced Pull Requests**: 
https://github.com/kiegroup/drools-wb/pull/1431

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: drools-scenario-simulation/drools-scenario-simulation-api/src/main/java/org/drools/scenariosimulation/api/model/ScesimModelDescriptor.java [hunk 1]
```diff
@@ -89,9 +89,9 @@ public int getIndexByIdentifier(FactIdentifier factIdentifier, ExpressionIdentif
     }
 
     public Optional<FactMapping> getFactMapping(FactIdentifier factIdentifier, ExpressionIdentifier ei) {
-        List<FactMapping> factMappings = internalFilter(e -> Objects.equals(e.getExpressionIdentifier(), ei) &&
+        List<FactMapping> filteredFactMappings = internalFilter(e -> Objects.equals(e.getExpressionIdentifier(), ei) &&
```
Comment (danielezonca) [inline drools-scenario-simulation/drools-scenario-simulation-api/src/main/java/org/drools/scenariosimulation/api/model/ScesimModelDescriptor.java]: It is not related to the PR but why don't make `internalFilter` returns `Stream<FactMapping>` (and update `getFactMappingsByFactName` accordingly)?
This should prevent useless iteration

**Review by jstastny-cz — APPROVED** [review]

File: drools-scenario-simulation/drools-scenario-simulation-api/src/main/java/org/drools/scenariosimulation/api/model/ScesimModelDescriptor.java (hunk 1 above)
Comment (yesamer) [reply drools-scenario-simulation/drools-scenario-simulation-api/src/main/java/org/drools/scenariosimulation/api/model/ScesimModelDescriptor.java]: @danielezonca Very nice, the only downside is we need to create a PR for drools-wb accordingly to that change (the API changes). Please wait for that PR before proceed.

**Comment by yesamer** [comment]
jenkins retest this please

File: drools-scenario-simulation/drools-scenario-simulation-api/src/main/java/org/drools/scenariosimulation/api/model/ScesimModelDescriptor.java (hunk 1 above)
Comment (danielezonca) [reply drools-scenario-simulation/drools-scenario-simulation-api/src/main/java/org/drools/scenariosimulation/api/model/ScesimModelDescriptor.java]: Actually I was proposing a more simple and limited change: make `internalFilter` returns a stream, then update `getFactMapping` as
```java
    public Optional<FactMapping> getFactMapping(FactIdentifier factIdentifier, ExpressionIdentifier ei) {
        return internalFilter(e -> Objects.equals(e.getExpressionIdentifier(), ei) &&
                Objects.equals(e.getFactIdentifier(), factIdentifier)).findFirst();
    }
```
And change
```java
    public List<FactMapping> getFactMappingsByFactName(String factName) {
        return internalFilter(e -> e.getFactIdentifier().getName().equalsIgnoreCase(factName)).collect(Collectors.toList());
    }
```
So in this case no impact on other repo :)

Comment (yesamer) [reply drools-scenario-simulation/drools-scenario-simulation-api/src/main/java/org/drools/scenariosimulation/api/model/ScesimModelDescriptor.java]: @danielezonca ok, in any case I think returning a Stream in  `getFactMappingsByFactName` works very well in this case, in client side the only case where that list is used is converted as a Stream and operation are used.

**Review by danielezonca — APPROVED** [review]
LGTM :+1:

**Comment by yesamer** [comment]
@danielezonca can you please merge it?

**Pull request merged**